test/unit_tests/graphql/conftest.py

```diff
@@ -38,7 +41,7 @@ class Metadata(BaseModel):
 
 @pytest.fixture(scope="session", autouse=True)
 def fix_graphql_model_registration():
-    internal_graphql_models = {}
+    internal_graphql_models = {"ProductBlockListNestedForTestInactive": ProductBlockListNestedForTestInactiveGraphql}
```

Comment:
small comment explaining why this one is hardcoded?

Reply:
something like this
```suggestion
    # This block caches the "ProductBlockListNestedForTestInactive" model to avoid re-instantiation in each test case.
    # This is necessary because this product block has a self referencing property, which strawberry can't handle correctly,
    # and lead to an error expecting the `ProductBlock` strawberry type to exist.
    internal_graphql_models = {"ProductBlockListNestedForTestInactive": ProductBlockListNestedForTestInactiveGraphql}
```

Reply:
👍